ppr-ui/src/utils/format-helper.ts

```diff
@@ -115,3 +115,42 @@ export function cleanEmpty<Type> (obj:Type): Type {
   })
   return newObj as Type
 }
+
+/**
+ * Formats a value into a currency string with optional dollar sign and specified number of decimal places.
+ *
+ * @param {string | number} value - The value to be formatted. Can be a string or number.
+ *                              Strings with dollar signs and commas will be converted to numbers.
+ *                              For example: '1,234.56' or '$1,234.56' will both be converted to 1234.56.
+ *
+ * @param {number} decimal - The number of decimal places to display. This sets both the minimum and maximum number of fraction digits.
+ *
+ * @param {boolean} dollarSign - Determines whether to include the CAD dollar sign in the formatted string.
+ *                             If true, will format as '$X,XXX.XX'. If false, will format as 'X,XXX.XX'.
+ *
+ * @returns {string} - Returns the formatted currency string.
+ *                   If the provided value is null, undefined, or an empty string, it returns an empty string.
+ *
+ * @example
+ * formatCurrency(1234.5678, 2, true)   // Returns '$1,234.57'
+ * formatCurrency('1,234.5678', 2, false) // Returns '1,234.57'
+ * formatCurrency('$1,234.5678', 3, true)  // Returns '$1,234.568'
+ */
+export function formatCurrency (value: string | number, decimal: number, dollarSign: boolean): string {
```

Comment:
Few suggestions here: 
- Can we add default values to `decimal` and `dollarSign`?  e.g. defaults to **2**, and **true**, because in both times that we use this function, those are the values, so we make them default (and if need be, they can be overwritten).
- For booleans, we should use the verb prefix like `is`, `has` or even `show`, etc. Could be `showDollarSign`?